File: python/algos/climb_stairs.py
def climb_stairs(stairs: int) -> int:
    """Solves stairs problem."""
    if stairs == 1:
        return 1
    elif stairs == 2:
        return 2
    else:
        last = 2
        earlier = 1
        answer = 0
        while stairs > 2:
            answer = earlier + last
            earlier = last
            last = answer
            stairs -= 1
        return answer


if __name__ == '__main__':
    assert climb_stairs(2) == 2
    assert climb_stairs(3) == 3
    assert climb_stairs(8) == 34
    assert climb_stairs(16) == 1597

File: rust/src/algos/climb_stairs.rs
pub fn climb_stairs(stairs: i32) -> i32 {
    if stairs == 1 {
        1
    } else if stairs == 2 {
        2
    } else {
        let mut n = stairs;
        let mut last = 2;
        let mut earlier = 1;
        let mut answer = 0;
        while n > 2 {
            answer = last + earlier;
            earlier = last;
            last = answer;
            n -= 1;
        }
        answer
    }
}
